test/contract_qvault.cpp

```diff
@@ -0,0 +1,875 @@
+    {
+        if(numberOfAuth == 1)
+        {
+            EXPECT_EQ(authAddress1, newAuthAddress);
+        }
+        else if(numberOfAuth == 2)
+        {
+            EXPECT_EQ(authAddress2, newAuthAddress);
+        }
+        else 
+        {
+            EXPECT_EQ(authAddress3, newAuthAddress);
+        }
+    }
+
+    void submitDistributionPermilleChecker(uint32 newQCAPHolderPt, uint32 newReinvestingPt, uint32 newDevPt)
+    {
+        EXPECT_EQ(newQCAPHolderPt, newQCAPHolderPermille1);
+        EXPECT_EQ(newQCAPHolderPt, newQCAPHolderPermille2);
+        EXPECT_EQ(newQCAPHolderPt, newQCAPHolderPermille3);
+
+        EXPECT_EQ(newReinvestingPt, newReinvestingPermille1);
+        EXPECT_EQ(newReinvestingPt, newReinvestingPermille2);
+        EXPECT_EQ(newReinvestingPt, newReinvestingPermille3);
+
+        EXPECT_EQ(newDevPt, newDevPermille1);
+        EXPECT_EQ(newDevPt, newDevPermille2);
+        EXPECT_EQ(newDevPt, newDevPermille3);
+    }
+
+    void changeDistributionPermilleChecker(uint32 newQCAPHolderPt, uint32 newReinvestingPt, uint32 newDevPt)
+    {
+        EXPECT_EQ(newQCAPHolderPt, QCAPHolderPermille);
+        EXPECT_EQ(newReinvestingPt, reinvestingPermille);
+        EXPECT_EQ(newDevPt, devPermille);
+    }
+
+    void submitReinvestingAddressChecker(id newReinvestingAddress)
+    {
+        EXPECT_EQ(newReinvestingAddress1, newReinvestingAddress);
+        EXPECT_EQ(newReinvestingAddress2, newReinvestingAddress);
+        EXPECT_EQ(newReinvestingAddress3, newReinvestingAddress);
+    }
+
+    void changeReinvestingAddressChecker(id newReinvestingAddress)
+    {
+        EXPECT_EQ(reinvestingAddress, newReinvestingAddress);
+    }
+
+    void submitAdminAddressChecker(id newAdminAddress)
+    {
+        EXPECT_EQ(newAdminAddress1, newAdminAddress);
+        EXPECT_EQ(newAdminAddress2, newAdminAddress);
+        EXPECT_EQ(newAdminAddress3, newAdminAddress);
+    }
+
+    void changeAdminAddressChecker(id newAdminAddress)
+    {
+        EXPECT_EQ(adminAddress, newAdminAddress);
+    }
+
+    void submitBannedAddressChecker(id newBannedAddress)
+    {
+        EXPECT_EQ(bannedAddress1, newBannedAddress);
+        EXPECT_EQ(bannedAddress2, newBannedAddress);
+        EXPECT_EQ(bannedAddress3, newBannedAddress);
+    }
+
+    void saveBannedAddressChecker(id newBannedAddress)
+    {
+        EXPECT_EQ(bannedAddress.get(numberOfBannedAddress - 1), newBannedAddress);
+    }
+
+    void submitUnbannedAddressChecker(id newUnbannedAddress)
+    {
+        EXPECT_EQ(unbannedAddress1, newUnbannedAddress);
+        EXPECT_EQ(unbannedAddress2, newUnbannedAddress);
+        EXPECT_EQ(unbannedAddress3, newUnbannedAddress);
+    }
+
+    void saveUnbannedAddressChecker(id unbannedAddress)
+    {
+        for(uint32 i = 0 ; i < numberOfBannedAddress; i++)
+        {
+            EXPECT_NE(unbannedAddress, bannedAddress.get(i));
+        }
+    }
+
+    void getDataChecker(getData_output output)
```

Comment:
You pass objects by value here and at other places in the test code (see parameters std::vector<id> and id).
Best practice is to pass objects by reference, which leads to faster code. I recommend to read 
https://stackoverflow.com/questions/70475034/c-best-practise-when-passing-in-arguments-to-functions and pass objects by cost reference if they are not changed in the function.

Reply:
I got it. fixed.